src/Aspire.Hosting/Dashboard/DashboardLifecycleHook.cs

```diff
@@ -214,15 +215,55 @@ private void ConfigureAspireDashboardResource(IResource dashboardResource)
                 }
             }
 
-            // Configure frontend browser token
-            if (!string.IsNullOrEmpty(browserToken))
+            // Configure frontend auth
+            context.EnvironmentVariables[DashboardConfigNames.DashboardFrontendAuthModeName.EnvVarName] = dashboardAuthMode.ToString();
+
+            if (dashboardAuthMode == DashboardAuthMode.OpenIdConnect)
             {
-                context.EnvironmentVariables[DashboardConfigNames.DashboardFrontendAuthModeName.EnvVarName] = "BrowserToken";
-                context.EnvironmentVariables[DashboardConfigNames.DashboardFrontendBrowserTokenName.EnvVarName] = browserToken;
+                if (options.OpenIdConnect != null)
+                {
+                    context.EnvironmentVariables[DashboardConfigNames.DashboardFrontendOpenIdConnectNameClaimType.EnvVarName] = options.OpenIdConnect.NameClaimType;
+                    context.EnvironmentVariables[DashboardConfigNames.DashboardFrontendOpenIdConnectUsernameClaimType.EnvVarName] = options.OpenIdConnect.UsernameClaimType;
+                    context.EnvironmentVariables[DashboardConfigNames.DashboardFrontendOpenIdConnectRequiredClaimType.EnvVarName] = options.OpenIdConnect.RequiredClaimType;
+                    context.EnvironmentVariables[DashboardConfigNames.DashboardFrontendOpenIdConnectRequiredClaimValue.EnvVarName] = options.OpenIdConnect.RequiredClaimValue;
+                }
+
+                if (options.OpenIdConnectSettings != null)
+                {
+                    var prefix = "AUTHENTICATION__SCHEMES__OPENIDCONNECT__";
```

Comment:
Originally it was, but [earlier feedback](https://github.com/dotnet/aspire/pull/6469#discussion_r1813962144) earlier feedback was to switch to POCO to avoid the dependency on it.  If there was a way to copy that entire section in then I guess you don't need the reference and could just copy that section in directly, but since there's currently no way to bring the whole section in I think POCO is a reasonable approach for now that doesn't really bind you in any way long term.

Reply:
@davidfowl Can you explain your earlier feedback? I don't like that it's using the same default configuration section `OpenIdConnectConfigureOptions` does but only supports a subset of the options supported by that. If the goal is just to remove Microsoft.AspNetCore.Authentication.OpenIdConnect PackageReference, I would just forward the entire "Authentication:Schemes:OpenIdConnect" `IConfigurationSection` via environment variables.